server/interpreter.js

```diff
@@ -3097,18 +3080,21 @@ Interpreter.prototype['stepUnaryExpression'] = function(stack, state, node) {
   } else if (node['operator'] === '~') {
     value = ~value;
   } else if (node['operator'] === 'delete') {
+    // If value is not an array, then it is a primitive.  If so, return true.
```

Comment:
This comment looks misplaced, but perhaps is actually mis-worded.  How about: "It's only possible to delete expressions which evaluate to a reference".